### PR TITLE
85 flowforge content securing node red

### PR DESCRIPTION
**Pull request by robmarcer**
## Description

Blog post discussing steps to secure Node-RED.

### CODE REVIEW

File: src/blog/2023/04/securing-node-red-in-production.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+A VPN provides a lot of security advantages depending on which you are using and how it is configured. To secure my traffic I'm going to use a great service call [Tailscale](https://tailscale.com/) which is free for personal projects.
+
+I'm going to install Tailscale on the Raspberry Pi I'm running Node-RED on as well as any other devices I want to access my project from. Once that's done I can access Node-RED from anywhere with internet access but more importantly the traffic to and from my devices is also encrypted.
+
+Before we start, it's important to remember that a VPN is only as secure as the company who runs it. You should always consider if you trust the VPN provider as they could potentially access your devices. I trust Tailscale but please do your own research before using a VPN provider.
```
Comment (iskerrett) [inline src/blog/2023/04/securing-node-red-in-production.md:96]: Is this really needed?

Comment (robmarcer) [reply src/blog/2023/04/securing-node-red-in-production.md:96]: It's not clear what 'this' is referring to, can you be specific please?

Comment (iskerrett) [reply src/blog/2023/04/securing-node-red-in-production.md:96]: The paragraph about trusting your vpn provider.

File: src/blog/2023/04/securing-node-red-in-production.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+Using these two features, we now have much better control over who can access the design interface for Node-RED.
+
+## Traffic to your Node-RED instance is unencrypted
+
+Hopefully, we all know the importance of encrypting your connections between devices to stop people intercepting your traffic. This isn't a huge concern when working on your home LAN but what if you want to access your Node-RED instance from a remote location? 
+
+There are two obvious options, HTTPS, and a VPN (Virtual Private Network).
+
+We could setup your Node-RED traffic to run over HTTPS, this solution ensures that all traffic to and from your Node-RED is encrypted. The downside to this approach is it's quite complex to set up. We will need to have a domain name, open up ports on our LAN's firewall, use a HTTPS certificate provider and then make sure we remember to renew the certificates as needed. It's doable if you are comfortable with those concepts (I covered how to do this as part of my blog [hosting FlowForge on Google Cloud](/blog/2022/12/flowforge-gcp-https-set-up/)) but there is an easier way to get started, using a VPN.
```
[the remark this answers is not in the document]
Comment (robmarcer) [reply src/blog/2023/04/securing-node-red-in-production.md:90]: Explaining how to set up HTTPs is so complex that it's at least an article in itself, probably one for each platform. The user needs to register a domain, setup DNS, allow traffic through a firewall to their device, register a cert (or get one from a free provider) then either manually renew their certs or setup something which can do it automatically for them.

If the goal was 'explain how to setup https on a Pi running Raspbian using Lets Encrypt' I'd expect it to be about a day's work. Happy to do so if that's valuable? I'd covered it for Docker on GCP in the linked article.

Furthermore, this only matters for ingress to Node-RED. I'd expect the vast majority of self hosted Node-RED cannot be accessed directly via the internet. It's complex to set up and can create security problems. These are not problems with Node-RED, it's the same with any web sever based application.

File: src/blog/2023/04/securing-node-red-in-production.md (hunk 1 above)
Comment (robmarcer) [reply src/blog/2023/04/securing-node-red-in-production.md:96]: I think so yes, I think we should recommend a VPN without stating the risks of using them (or any VPN provider).

Comment (iskerrett) [reply src/blog/2023/04/securing-node-red-in-production.md:96]: OK, it just sounds a bit patronizing to me but your call.

**Pull request merged**